packages/studio-be/src/core/bots/bot-service.ts

```diff
@@ -285,6 +286,7 @@ export class BotService {
     }
 
     await this.cms.clearElementsFromCache(botId)
+    BOTPRESS_EVENTS.emit('unmountBot', { botId })
```

Comment:
This is so hackish...

There's a lot of ways of doing this, but this one just feels off. I personally would do something like:

**solution 1**
```ts
class BotService {
    public listenForBotUnmount(listener: (botId: string) => Promise<void>) {}
}

class FlowService {
    constructor(private botService: BotService) {
        botService.listenForBotUnmount(this.handleUnmount.bind(this))
    }
    
    private handleUnmount(botId: string) { delete this.scopes[botId] }
}
```

**solution 2**
```ts
class BotService {
    constructor(private flowService: FlowService) {}
    
    public unmountBot(botId: string) {
        this.flowService.removeFlowsForBot(botId)
    }
}

class FlowService { 
    public removeFlowsForBot(botId: string) { delete this.scopes[botId] }
}
```